kbase-extension/static/kbase/js/widgets/function_output/kbaseMatrix2DAbstract.js

```diff
@@ -0,0 +1,506 @@
+            var style = 'lftip';
+            if(data.length <= iDisplayLength) { style = 'fti'; }
+
+            var table = $('<table>')
+                .attr('id', pref+'conditions-table')
+                .addClass('table table-bordered table-striped')
+                .css('width', '100%')
+                .css('margin-left', '0px')
+                .css('margin-right', '0px')
+                .appendTo($container)
+                .dataTable( {
+                    "sDom": style,
+                    "iDisplayLength": iDisplayLength,
+                    "aaData": data,
+                    "aoColumns": columns,
+                    "oLanguage": {
+                                "sEmptyTable": emptyTableMessage,
+                                "sSearch": "Search: "
+                    }                    
+                } );              
+            
+        },    
+        
+        buildMatrixOverview: function($container){
+            var pref = self.pref;
+            
+            
+            var $tableOverview = $('<table>')
+                .attr('id',pref+'overview-table')
+                .addClass('table table-striped table-bordered')
+                .css('width','100%')
+                .css('margin-left','0px')
+                .css('margin-right','0px')
+                .appendTo($container);            
+                                
+            for(var i in this.matrix.metadata.matrix_metadata){
+                var md = this.matrix.metadata.matrix_metadata[i];
+                
+                var label = md.category;
+                if( md.property_name != undefined && md.property_name != ''){
+                    label += "." + md.property_name;
+                }
+                
+                $tableOverview.append( this.makeRow( 
+                    label, 
+                    md.property_value + (md.property_unit ? md.property_unit : "" ) ) );                
+            }            
+        },           
+  
+        makeRow: function(name, value) {
+            var $row = $("<tr/>")
+                       .append($("<th />").css('width','20%').append(name))
+                       .append($("<td />").append(value));
+            return $row;
+        },
+        
+        loading: function(isLoading) {
+            if (isLoading)
+                this.showMessage("<img src='" + this.options.loadingImage + "'/>");
+            else
+                this.hideMessage();                
+        },
+
+        showMessage: function(message) {
+            var span = $("<span/>").append(message);
+
+            this.$messagePane.append(span);
+            this.$messagePane.show();
+        },
+
+        hideMessage: function() {
+            this.$messagePane.hide();
+            this.$messagePane.empty();
+        },
+
+        clientError: function(error){
+            this.loading(false);
+            this.showMessage(error.error.error);
+        },            
+
+        uuid: function() {
```

Comment:
There's a utility module you can use for uuids (and some other things) in util/string. See https://github.com/kbase/narrative/blob/develop/docs/developer/widget_helper_modules.md#utilstring for an example.